sql-plugin/src/main/301until310-nondb/scala/com/nvidia/spark/rapids/shims/v2/GpuBroadcastHashJoinExec.scala

```diff
@@ -68,22 +113,40 @@ class GpuBroadcastHashJoinMeta(
 
   override def convertToGpu(): GpuExec = {
     val Seq(left, right) = childPlans.map(_.convertIfNeeded())
+    val gpuBuildSide = GpuJoinUtils.getGpuBuildSide(join.buildSide)
     // The broadcast part of this must be a BroadcastExchangeExec
     val buildSide = join.buildSide match {
       case BuildLeft => left
       case BuildRight => right
     }
     verifyBuildSideWasReplaced(buildSide)
-    val joinExec = GpuBroadcastHashJoinExec(
-      leftKeys.map(_.convertToGpu()),
-      rightKeys.map(_.convertToGpu()),
-      join.joinType,
-      GpuJoinUtils.getGpuBuildSide(join.buildSide),
-      None,
-      left, right)
-    // The GPU does not yet support conditional joins, so conditions are implemented
-    // as a filter after the join when possible.
-    condition.map(c => GpuFilterExec(c.convertToGpu(), joinExec)).getOrElse(joinExec)
+
+    val condition = conditionMeta.map(_.convertToGpu())
+    val substituteBroadcastNestedLoopJoin: Boolean = join.joinType match {
+      case RightOuter | LeftOuter | LeftSemi | LeftAnti  =>
+        isAstCondition()
+      case _ => false
+    }
+
+    if (substituteBroadcastNestedLoopJoin) {
+      val joinExec = ShimLoader.getSparkShims.getGpuBroadcastNestedLoopJoinShim(
```

Comment:
could just return here, without defining `joinExec`)

Reply:
Good point.  I will change this.